packages/dds/tree/src/simple-tree/proxies.ts

```diff
@@ -47,6 +47,24 @@ import { tryGetFlexNodeTarget, setFlexNode, getFlexNode, tryGetFlexNode } from "
 import { cursorFromFieldData, cursorFromNodeData } from "./toMapTree.js";
 import { RawTreeNode, createRawNode, extractRawNodeContent } from "./rawNode.js";
 
+/**
+ * Detects if the given 'candidate' is a TreeNode.
+ *
+ * @remarks
+ * Supports both Hydrated and {@link Unhydrated} TreeNodes, both of which return true.
+ *
+ * Because the common usage is to check if a value being inserted/set is a TreeNode,
+ * this function permits calling with primitives as well as objects.
+ *
+ * Primitives will always return false (as they are copies of data, not references to nodes).
+ *
+ * @param candidate - Value which may be a TreeNode
+ * @returns true if the given 'candidate' is a hydrated TreeNode.
+ */
+export function isTreeNode(candidate: unknown): candidate is TreeNode | Unhydrated<TreeNode> {
```

Comment:
We don't expose `TreeNode`s as a part of the public API, right? So this has no impact on user workflows?

Reply:
I guess we don't expect users to need to make this sort of check. That's really more what I wanted to verify.

Reply:
We might want/need to export it at some point, but not for now.